modules/gateway/nodes.go

```diff
@@ -53,14 +54,9 @@ func (g *Gateway) managedAddUntrustedNode(addr modules.NetAddress) error {
 	//
 	// NOTE: this is a somewhat clunky way of specifying that you didn't
 	// actually want a connection.
-	encoding.WriteObject(conn, "0.0.0")
-	var reject string
-	err = encoding.ReadObject(conn, &reject, build.MaxEncodedVersionLength)
-	if err != nil {
-		g.log.Debugln("ERROR: version handshake ping terminated unexpectedly:", err)
-	}
-	if reject != "reject" {
-		g.log.Debugln("WARN: peer does not seem to have correctly rejected our ping:", reject)
+	_, err = connectVersionHandshake(conn, "0.0.0")
+	if err != errPeerRejectedConn {
```

Comment:
during the version handshake, first we send our version ("0.0.0"), and then the peer replies with their version. If our version is unacceptable to the peer (and it always will be, unless they are running custom code), they will reply with the string "reject" instead of their actual version. The connecting node recognizes this as a special case and returns `errPeerRejectedConn`.

Technically the peer can reject our handshake for any reason, not just because our version was bad. For example, they could reject new peers instead of kicking out old ones. I think the variable name is ok for now.